Tokens: Unable to view metadata for some of my existing tokens

In tools, I am trying to view a token that I created a few weeks ago. I'm not looking for the token value, but just wanted to see what it would display in the UI.  When I select the token, which is a read-only token to one of my buckets, I get an internal error. In the console I see this message. This particular token didn't have any telegraf references.

```
RequiredError: Required parameter telegrafID was null or undefined when calling getTelegrafsID.
    at new t (index.mjs:1)
    at Object.getTelegrafsID (index.mjs:1)
    at Object.getTelegrafsID (index.mjs:1)
    at t.getTelegrafsID (index.mjs:1)
    at e.<anonymous> (index.mjs:1)
    at index.mjs:1
    at Object.next (index.mjs:1)
    at index.mjs:1
    at new Promise (<anonymous>)
    at a (index.mjs:1)
(anonymous) @ thunks.ts:203
await in (anonymous) (async)
(anonymous) @ index.js:16
n.<computed> @ bindActionCreators.js:9
S @ EditTokenOverlay.tsx:77
await in S (async)
(anonymous) @ EditTokenOverlay.tsx:62
Tu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
ku @ react-dom.production.min.js:261
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
2react-dom.production.min.js:216 TypeError: n.forEach is not a function
    at s (IndividualAccordionBody.tsx:36)
    at na (react-dom.production.min.js:157)
    at zs (react-dom.production.min.js:267)
    at Au (react-dom.production.min.js:250)
    at _u (react-dom.production.min.js:250)
    at wu (react-dom.production.min.js:250)
    at du (react-dom.production.min.js:243)
    at react-dom.production.min.js:123
    at t.unstable_runWithPriority (scheduler.production.min.js:18)
    at Uo (react-dom.production.min.js:122)
is @ react-dom.production.min.js:216
i.componentDidCatch.n.callback @ react-dom.production.min.js:217
ci @ react-dom.production.min.js:131
ds @ react-dom.production.min.js:220
Su @ react-dom.production.min.js:259
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Eu @ react-dom.production.min.js:252
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
```

When I tried to access another token (which has read/write access to a particular bucket) I get this error: 

```
generatedRoutes.ts:2309 GET https://influxdb.aws.influxdata.io/api/v2/buckets/undefined/schema/measurements 400
a.request @ generatedRoutes.ts:2309
bn @ generatedRoutes.ts:8777
(anonymous) @ thunks.ts:186
(anonymous) @ index.js:16
n.<computed> @ bindActionCreators.js:9
S @ EditTokenOverlay.tsx:84
(anonymous) @ EditTokenOverlay.tsx:62
Tu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
ku @ react-dom.production.min.js:261
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
thunks.ts:196 error while retrieving schemas Error: corrupt ID provided: id must have a length of 16 bytes
    at thunks.ts:191
    at async S (EditTokenOverlay.tsx:84)
(anonymous) @ thunks.ts:196
await in (anonymous) (async)
(anonymous) @ index.js:16
n.<computed> @ bindActionCreators.js:9
S @ EditTokenOverlay.tsx:84
(anonymous) @ EditTokenOverlay.tsx:62
Tu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
ku @ react-dom.production.min.js:261
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
generatedRoutes.ts:2309 GET https://influxdb.aws.influxdata.io/api/v2/buckets/undefined/schema/measurements 400
a.request @ generatedRoutes.ts:2309
bn @ generatedRoutes.ts:8777
(anonymous) @ thunks.ts:186
(anonymous) @ index.js:16
n.<computed> @ bindActionCreators.js:9
S @ EditTokenOverlay.tsx:84
await in S (async)
(anonymous) @ EditTokenOverlay.tsx:62
Tu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
ku @ react-dom.production.min.js:261
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
thunks.ts:196 error while retrieving schemas Error: corrupt ID provided: id must have a length of 16 bytes
    at thunks.ts:191
    at async S (EditTokenOverlay.tsx:84)
(anonymous) @ thunks.ts:196
await in (anonymous) (async)
(anonymous) @ index.js:16
n.<computed> @ bindActionCreators.js:9
S @ EditTokenOverlay.tsx:84
await in S (async)
(anonymous) @ EditTokenOverlay.tsx:62
Tu @ react-dom.production.min.js:262
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
ku @ react-dom.production.min.js:261
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
react-dom.production.min.js:216 TypeError: n.forEach is not a function
    at s (IndividualAccordionBody.tsx:36)
    at na (react-dom.production.min.js:157)
    at zs (react-dom.production.min.js:267)
    at Au (react-dom.production.min.js:250)
    at _u (react-dom.production.min.js:250)
    at wu (react-dom.production.min.js:250)
    at du (react-dom.production.min.js:243)
    at react-dom.production.min.js:123
    at t.unstable_runWithPriority (scheduler.production.min.js:18)
    at Uo (react-dom.production.min.js:122)
is @ react-dom.production.min.js:216
i.componentDidCatch.n.callback @ react-dom.production.min.js:217
ci @ react-dom.production.min.js:131
ds @ react-dom.production.min.js:220
Su @ react-dom.production.min.js:259
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Eu @ react-dom.production.min.js:252
du @ react-dom.production.min.js:243
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:18
Uo @ react-dom.production.min.js:122
Vo @ react-dom.production.min.js:123
Wo @ react-dom.production.min.js:122
Re @ react-dom.production.min.js:292
Qt @ react-dom.production.min.js:73
E.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:495
```

--- a/src/authorizations/components/redesigned/EditResourceAccordion.tsx
+++ b/src/authorizations/components/redesigned/EditResourceAccordion.tsx
@@ -8,11 +8,16 @@ import {ResourceAccordionHeader} from 'src/authorizations/components/redesigned/
 import {IndividualAccordionBody} from 'src/authorizations/components/redesigned/IndividualAccordionBody'
 import {AllAccordionBody} from './AllAccordionBody'
 import {formatResources} from 'src/authorizations/utils/permissions'
+import FilterList from 'src/shared/components/FilterList'
 
+// Types
+import {Resource} from 'src/client'
 interface Props {
   permissions: any
+  searchTerm: string
 }
 
+const Filter = FilterList<Resource>()
 export class EditResourceAccordion extends Component<Props> {
   public render() {
     const {permissions} = this.props
@@ -48,44 +53,83 @@ export class EditResourceAccordion extends Component<Props> {
         {!isEmpty(allResourceNames[1]) && (
           <Accordion key="Other Resources" expanded={true}>
             <ResourceAccordionHeader resourceName="Other Resources" />
-            {allResourceNames[1].map(resource => {
-              const resourceName =
-                resource.charAt(0).toUpperCase() + resource.slice(1)
-
-              return (
-                <AllAccordionBody
-                  key={resource}
-                  resourceName={resourceName}
-                  permissions={permissions[resource]}
-                  disabled={true}
-                />
-              )
-            })}
+            {this.getOtherResourceAccordionBody(allResourceNames)}
           </Accordion>
         )}
       </>
     )
   }
 
+  getOtherResourceAccordionBody = allResourceNames => {
+    const {permissions, searchTerm} = this.props
+    const resourcePermissions = []
+    allResourceNames[1].forEach(resource => {
+      resourcePermissions.push({name: resource, perm: permissions[resource]})
+    })
+    return (
+      <Filter
+        list={resourcePermissions}
+        searchTerm={searchTerm}
+        searchKeys={['name']}
+      >
+        {filteredNames => (
+          <AllAccordionBody
+            resourceName="All Resources"
+            permissions={filteredNames}
+            disabled={true}
+          />
+        )}
+      </Filter>
+    )
+  }
+
   getAccordionBody = (resourceName, resource) => {
-    const {permissions} = this.props
+    const {permissions, searchTerm} = this.props
+    const permissionNames = []
+
     if (resourceName === 'Telegrafs') {
+      for (const [, value] of Object.entries(
+        permissions[resource].sublevelPermissions
+      )) {
+        permissionNames.push(value)
+      }
       return (
-        <IndividualAccordionBody
-          resourceName={resource}
-          permissions={permissions[resource].sublevelPermissions}
-          title="Individual Telegraf Configuration Names"
-          disabled={true}
-        />
+        <Filter
+          list={permissionNames}
+          searchTerm={searchTerm}
+          searchKeys={['name']}
+        >
+          {filteredNames => (
+            <IndividualAccordionBody
+              resourceName={resource}
+              permissions={filteredNames}
+              title="Individual Telegraf Configuration Names"
+              disabled={true}
+            />
+          )}
+        </Filter>
       )
     } else if (resourceName === 'Buckets') {
+      for (const [, value] of Object.entries(
+        permissions[resource].sublevelPermissions
+      )) {
+        permissionNames.push(value)
+      }
       return (
-        <IndividualAccordionBody
-          resourceName={resource}
-          permissions={permissions[resource].sublevelPermissions}
-          title="Individual Bucket Names"
-          disabled={true}
-        />
+        <Filter
+          list={permissionNames}
+          searchTerm={searchTerm}
+          searchKeys={['name']}
+        >
+          {filteredNames => (
+            <IndividualAccordionBody
+              resourceName={resource}
+              permissions={filteredNames}
+              title="Individual Bucket Names"
+              disabled={true}
+            />
+          )}
+        </Filter>
       )
     }
   }

--- a/src/authorizations/components/redesigned/EditTokenOverlay.tsx
+++ b/src/authorizations/components/redesigned/EditTokenOverlay.tsx
@@ -21,6 +21,7 @@ import {
   Page,
 } from '@influxdata/clockface'
 import {EditResourceAccordion} from 'src/authorizations/components/redesigned/EditResourceAccordion'
+import SearchWidget from 'src/shared/components/search_widget/SearchWidget'
 
 // Types
 import {Authorization} from 'src/types'
@@ -56,6 +57,7 @@ const EditTokenOverlay: FC<Props> = props => {
   )
   const [label, setlabel] = useState(props.auth.status)
   const [permissions, setPermissions] = useState([])
+  const [searchTerm, setSearchTerm] = useState('')
 
   useEffect(() => {
     if (_.isEmpty(permissions)) {
@@ -98,6 +100,10 @@ const EditTokenOverlay: FC<Props> = props => {
   const handleInputChange = event => {
     setDescription(event.target.value)
     setStatus(ComponentStatus.Default)
+  }
+
+  const handleChangeSearchTerm = (searchTerm: string): void => {
+    setSearchTerm(searchTerm)
   }
 
   const handleDismiss = () => props.onDismissOverlay()
@@ -160,6 +166,11 @@ const EditTokenOverlay: FC<Props> = props => {
             </FlexBox>
           </Form>
           <FlexBox.Child className="main-flexbox-child">
+            <SearchWidget
+              searchTerm={searchTerm}
+              placeholderText="Filter Access Permissions..."
+              onSearch={handleChangeSearchTerm}
+            />
             <FlexBox
               margin={ComponentSize.Large}
               justifyContent={JustifyContent.SpaceBetween}
@@ -192,6 +203,7 @@ const EditTokenOverlay: FC<Props> = props => {
             </FlexBox>
             <EditResourceAccordion
               permissions={formatPermissionsObj(permissions)}
+              searchTerm={searchTerm}
             />
           </FlexBox.Child>
           <Page.ControlBarCenter>

